https://issues.redhat.com/browse/ACM-14268 (#6974)
open-cluster-management/rhacm-docs: business_continuity/backup_restore/backup_existing_clusters.adoc

[#backup-existing-clusters]

> = Understanding outcomes for existing managed clusters during the restore process

Reviewer: When I looked at the Google doc, the title is "Using an existing hub as a restore hub"

But should read as: "Using an existing hub cluster as a restore hub cluster"

It seems like this should be changed back to the title. Apologies if my suggestions in the first review were misleading. But should read as: "Using an existing hub cluster as a restore hub cluster". I was just confused before because I thought it was intended to be a task file.

> When you use a restore hub cluster with resources that a user created before running the restore operation, you can tag user-created resources on the restore hub cluster with the `velero.io/backup-name: backupName` label. If you do, understand the following constraints: 

Reviewer: Suggesting to organize this more cohesively based on the Tone guidance: https://www.ibm.com/docs/en/ibm-style?topic=medium-tone 

From my perspective, the intention of this file is to help users know what to expect if they decide to restore existing managed clusters. The statement, "When you use a restore hub cluster with resources that a user created before running the restore operation, you can tag user-created resources on the restore hub cluster with the `velero.io/backup-name: backupName` label." seems out of place. In my later suggestions, you can see i recommend to add it in the Tagging resources file.

```suggestion
Before you run the restore operation for existing hub clusters, understand the following constraints: 
```

> * The existing managed cluster can be detached during the restore process.

Reviewer: Why do we say "can be"? It makes it seem like a task for the user but it's a constraint
```suggestion
* Existing managed cluster are detached during the restore process.
```

> * The restore hub cluster can have more resources than the backup hub cluster.

Reviewer: Based on my previous question:
```suggestion
* There are more resources on the restore hub cluster than the backup hub cluster.
```

[#existing-clusters-detached]

> == Existing managed clusters can be detached during the restore process

Reviewer: ```suggestion
== Existing managed clusters are detached during the restore process
```

If the hub you plan to use for the restore operation already manages some clusters, expect the following outcomes when you run a restore operation with the `cleanupBeforeRestore` option set to `CleanupRestored`:  

* If an earlier backup created the `ManagedCluster` resources, then the `velero.io/backup-name: backupName` label annotation is set. Therefore, if you run a restore operation and use the `cleanupBeforeRestore=CleanupRestored` option, the clusters are detached and the data is cleaned up on the managed clusters. This only occurs if the backup managed clusters resources are not part of the restored backup.

> * If the `ManagedCluster` resource does not have the `velero.io/backup-name: backupName` label annotation, and you run a restore operation with the `cleanupBeforeRestore` option set to `CleanupRestored`, then the managed cluster resources are not cleaned up since they were not created by an earlier backup. Therefore, the managed clusters continue to be managed by this hub cluster, even though the clusters were not managed by the backup hub cluster, and this hub does not match the content of the backup hub cluster.

Reviewer: This is also related to the Tone guidance and the descriptions were a little confusing for me. The statement, "if you run a restore operation and use the `cleanupBeforeRestore=CleanupRestored` option.." was repeated in both bullet points so I thought it might be best to mention it once to simplify the descriptions. 

```suggestion
When you run a restore operation on your hub cluster that already manages some clusters and the `cleanupBeforeRestore` option set to `CleanupRestored`, the following outcomes occur:  

* Only when an earlier backup creates the `ManagedCluster` resources, the `velero.io/backup-name: backupName` label annotation is set, and the backup managed clusters are not part of the restored backup, managed clusters are detached and the data is cleaned on the managed clusters.
 
* When the `ManagedCluster` resources are not created by an earlier backup and do not have the `velero.io/backup-name: backupName` label annotation, `ManagedCluster` resources are not cleaned up. The managed clusters continue to be managed by your hub cluster and does not match the content of the backup hub cluster.
```

Does "detach" mean that the content of a managed clusters do not match the contents of the backup hub cluster? If yes, maybe we should start with explaining what detached is since we told the user that's what occurs. We can remove the explanation from the last bullet point.

```suggestion
A detached managed cluster has content that does not match the content of the backup hub cluster. When you run a restore operation on your hub cluster that already manages some clusters and the `cleanupBeforeRestore` option set to `CleanupRestored`, the following outcomes occur:  

* Only when an earlier backup creates the `ManagedCluster` resources, the `velero.io/backup-name: backupName` label annotation is set, and the backup managed clusters are not part of the restored backup, managed clusters are detached and the data is cleaned on the managed clusters.
 
* When the `ManagedCluster` resources are not created by an earlier backup and do not have the `velero.io/backup-name: backupName` label annotation, `ManagedCluster` resources are not cleaned up. The managed clusters continue to be managed by your hub cluster.
```

[#restore-hub-resources]

> == Restore hub can have more resources than the backup hub

Reviewer: ```suggestion
== Restore hub cluster has more resources than the backup hub cluster
```

If your restore operation uses a hub with earlier created user data, then the restore operation results in a hub that does not have the exact content as the primary hub. The user data includes the following resources: `policy`, `application`, `credential`, or any custom resource defined by the user. A restore operation with the `cleanupBeforeRestore` option set to `CleanupRestored` is not going to clean up these resources because they were created by the users and not created by an earlier backup. 

> If you intend to use an existing hub for a restore hub configuration and expect that the hub has the same content as the backup hub after the restore is completed, tag any resource that would be part of a hub backup with the `velero.io/backup-name: backupName` label annotation. To do so, complete the following documentation: xref:../backup_restore/backup_tag_resources.adoc#backup-tagging-resources[Tagging resources].

Reviewer: Should the second paragraph mention managed cluster instead of hub cluster? I just thought yes because this file mentions understanding outcomes for "existing managed clusters". Maybe also consider placeing 

```suggestion
When you restore a hub cluster with earlier user-created data and the `cleanupBeforeRestore` option is set to `CleanupRestored` for the restore operation, the resources of your restore hub cluster do not match the resources of the primary hub cluster. The user data includes the following resources: `policy`, `application`, `credential`, or any custom resource defined by the user. The resources are not cleaned because they are created by the users and not created by an earlier backup.

If you intend to use an existing hub cluster for a restore hub cluster configuration and expect for the hub cluster to have the same content as the backup hub cluster after the restore is completed, tag any resource that you want to be part of a hub cluster backup. See xref:../backup_restore/backup_tag_resources.adoc#backup-tagging-resources[Tagging resources] for more information.
```

> [#backup-existing-clusters]

Reviewer 2: @jc-berger I am sorry if I missed this in the initial review .. 

This section talks about : Using an existing hub cluster as a restore hub cluster , so the link should be matching somehow the content .. it is not about backing up existing clusters
I would probably use : #restore-using-existing-clusters

> = Understanding outcomes for existing managed clusters during the restore process

Reviewer 2: This title is not matching what we are talking about here, I don't remember seeing it on the first review

Should this be : Using an existing hub cluster as a restore hub cluster ?

> Before you run the restore operation for existing hub clusters, understand the following constraints: 

Reviewer 2: I find this confusing, I don't remember seeing it on the initial review ( Before you run the restore operation for existing hub clusters, understand the following constraints: )
The point I tried to make : If you use as a restore hub a cluster with resources you created before the restore operation, you must understand the following constraints: etc )
So I want to say: if you run a restore operation and use a hub cluster with some user data already created on this cluster, then you must be aware that : 
1. Existing managed cluster can be detached during the restore process
2. Restore hub can have more resources than the backup hub
